src/typings/cucumber.d.ts

```diff
@@ -0,0 +1,216 @@
+		}
+
+		interface FeaturesResultPayload extends EventPayload {
+			getDuration(): any;
+			getScenarioCounts(): any;
+			getStepCounts(): any;
+			isSuccessful(): boolean;
+		}
+
+		interface FeaturePayload extends EventPayload {
+			getStepKeywordByLines(): any;
+			getScenarioKeyword(): string;
+			getKeyword(): string;
+			getName(): string;
+			getDescription(): string;
+			getUri(): string;
+			getLine(): number;
+			getTags(): Tag[];
+			getScenarios(): ScenarioPayload[];
+			getPayloadItem(): FeaturePayload;
+		}
+
+		interface ScenarioPayload extends EventPayload {
+			getName(): string;
+			getKeyword(): string;
+			getDescription(): string;
+			getFeature(): FeaturePayload;
+			getUri(): string;
+			getUris(): string[];
+			getLine(): number;
+			getLines(): number[];
+			getTags(): Tag[];
+			getSteps(): any[];
+			getPayloadItem(): ScenarioPayload;
+		}
+
+		interface ScenarioResultPayload extends EventPayload {
+			getFailureException(): Error;
+			getScenario(): any;
+			getStatus(): any;
+		}
+
+		interface StepPayload extends EventPayload {
+			isHidden(): boolean;
+			isOutlineStep(): boolean;
+			getKeyword(): string;
+			getName(): string;
+			hasUri(): boolean;
+			getUri(): string;
+			getLine(): number;
+			getPreviousStep(): any;
+			hasPreviousStep(): boolean;
+			getAttachment(): any;
+			getAttachmentContents(): any;
+			getDocString(): string;
+			getDataTable(): any;
+			hasAttachment(): boolean;
+			hasDocString(): boolean;
+			hasDataTable(): boolean;
+			ensureDataTableIsAttached(): void;
+			isOutcomeStep(): boolean;
+			isEventStep(): boolean;
+			hasOutcomeStepKeyword(): boolean;
+			hasEventStepKeyword(): boolean;
+			isRepeatingOutcomeStep(): boolean;
+			isRepeatingEventStep(): boolean;
+			hasRepeatStepKeyword(): boolean;
+			isPrecededByOutcomeStep(): boolean;
+			isPrecededByEventStep(): boolean;
+		}
+
+		interface StepResultPayload extends EventPayload {
+			getAmbiguousStepDefinitions(): any[];
+			getAttachments(): any[];
+			getDuration(): any;
+			getFailureException(): Error;
+			getStep(): any;
+			getStepDefinition(): any;
+			getStatus(): any;
+			hasAttachments(): boolean;
+		}
+
+	}
+
+	interface Tag {
+		getName(): string;
+		getLine(): number;
+	}
+
+	export interface Scenario {
+		getKeyword(): string;
+		getName(): string;
+		getDescription(): string;
+		getUri(): string;
+		getLine(): number;
+		getTags(): Tag[];
+		getException(): Error;
+		getAttachments(): any[];
+		attach(data: any, mimeType?: string, callback?: (err?: any) => void): void;
+		isSuccessful(): boolean;
+		isFailed(): boolean;
+		isPending(): boolean;
+		isUndefined(): boolean;
+		isSkipped(): boolean;
+	}
+
+	interface EventHook {
+		(event: events.Event, callback?: ()=>void): void;
+	}
+
+	export interface SupportCodeConsumer {
+		(stepDefinitions:StepDefinitions | Hooks):void;

```

Comment:
Yeah, you're right. I understand now.
Sorry, I'm still fairly new to Typescript.
Thanks.